### PR TITLE
Ticket3949 tests

### PR DESCRIPTION
**Pull request by FreddieAkeroyd**
### Description of work

Add tests for run control (which test gencontrol) and also alerts

### Ticket

See ISISComputingGroup/IBEX#3949

### Acceptance criteria

*List the acceptance criteria for the PR. The aim is provide information to help the reviewer*

### Unit tests

*Give an overview of unit tests you have added or modified, if applicable. The aim is provide information to help the reviewer*

### System tests

*Mention any automated tests or manual tests that you have added or modified, if applicable. The aim is provide information to help the reviewer*

### Documentation
*Highlight and provide a link to any additions or changes to the documentation, if applicable. The aim is provide information to help the reviewer*

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: test_genie_python_using_simple.py [hunk 1]
```diff
@@ -270,4 +270,80 @@ def test_GIVEN_waiting_for_value_above_low_limit_on_block_WHEN_block_reaches_lim
...
+        g.set_pv(pw_pv, "dummy")
+        g.set_pv(inst_pv, "TESTINST")
+        g.set_pv(url_pv, "test") # this needs to be "test"
+        g.set_pv(self.block_pv + ":AC:LOW", 1)
+        g.set_pv(self.block_pv + ":AC:HIGH", 2)
+        g.set_pv(self.block_pv + ":AC:ENABLE", 1)
+        time.sleep(5)
+        assert_that(g.get_pv(out_pv), is_(1))
+
+    def _waitfor_runstate(self, state):
```
Comment (DominicOram) [inline test_genie_python_using_simple.py]: Can you add a comment here as to why you're not using waitfor_runstate?

File: test_genie_python_using_simple.py [hunk 2]
```diff
@@ -270,4 +270,80 @@ def test_GIVEN_waiting_for_value_above_low_limit_on_block_WHEN_block_reaches_lim
...
+        g.cset(self.block_name, runcontrol=False)
+        self.block_pv = g.prefix_pv_name("CS:SB:") + self.block_name
+        g.set_pv(self.block_pv + ":AC:ENABLE", 0)
+        self._waitfor_runstate("SETUP")
+
+    def tearDown(self):
+        g.abort()
+
+    def test_GIVEN_out_of_range_block_WHEN_start_run_THEN_dae_waiting(self):
+        if g.get_runstate() != "SETUP":
```
Comment (DominicOram) [inline test_genie_python_using_simple.py]: This check is already done at the end of setUp, is it required here?

File: test_genie_python_using_simple.py [hunk 3]
```diff
@@ -270,4 +270,80 @@ def test_GIVEN_waiting_for_value_above_low_limit_on_block_WHEN_block_reaches_lim
...
+
+class TestRunControl(unittest.TestCase):
+
+    def setUp(self):
+        g.set_instrument(None)
+        load_config_if_not_already_loaded(SIMPLE_CONFIG_NAME)
+        self.block_name = "FLOAT_BLOCK"
+        self.wait_before = 1
+        self.wait_after = 2
+        self.max_wait = (self.wait_before + self.wait_after) * 2
```
Comment (DominicOram) [inline test_genie_python_using_simple.py]: Are these wait variables used?

File: test_genie_python_using_simple.py (hunk 2 above)
Comment (FreddieAkeroyd) [reply test_genie_python_using_simple.py]: I think i was having some issues at some point which is why it was here, or else it was copied from somewhere else. I'll remove.

File: test_genie_python_using_simple.py (hunk 3 above)
Comment (FreddieAkeroyd) [reply test_genie_python_using_simple.py]: No they are not - copied from an earlier test

File: test_genie_python_using_simple.py (hunk 2 above)
Comment (FreddieAkeroyd) [reply test_genie_python_using_simple.py]: test_genie_python_dae does it, but it didn't check in setup

File: test_genie_python_using_simple.py (hunk 1 above)
Comment (FreddieAkeroyd) [reply test_genie_python_using_simple.py]: No reason, just a copy of logic done already, will change

**Pull request merged**